src/reducers/columnStateHelper.js

```diff
@@ -0,0 +1,359 @@
+  }
+
+  scrollX = oldState ? oldState.scrollX : 0;
+  if (props.scrollLeft !== undefined) {
+    scrollX = props.scrollLeft;
+  }
+
+  var groupHeaderHeight = useGroupHeader ? props.groupHeaderHeight : 0;
+
+  var columnResizingData;
+  if (props.isColumnResizing) {
+    columnResizingData = oldState && oldState.columnResizingData;
+  } else {
+    columnResizingData = EMPTY_OBJECT;
+  }
+
+  var columns;
+  var columnGroups;
+
+  if (useGroupHeader) {
+    var columnGroupSettings =
+      FixedDataTableWidthHelper.adjustColumnGroupWidths(
+        children,
+        props.width
+    );
+    columns = columnGroupSettings.columns;
+    columnGroups = columnGroupSettings.columnGroups;
+  } else {
+    columns = FixedDataTableWidthHelper.adjustColumnWidths(
+      children,
+      props.width
+    );
+  }
+
+  var columnInfo = _populateColumnsAndColumnData(
+    columns,
+    columnGroups,
+    oldState
+  );
+
+  var lastScrollToColumn = oldState ? oldState.scrollToColumn : undefined;
+  if (props.scrollToColumn !== null && props.scrollToColumn !== lastScrollToColumn) {
+    // If selected column is a fixed column, don't scroll
+    var fixedColumnsCount = columnInfo.bodyFixedColumns.length;
+    if (props.scrollToColumn >= fixedColumnsCount) {
+      var totalFixedColumnsWidth = 0;
+      var i, column;
+      for (i = 0; i < columnInfo.bodyFixedColumns.length; ++i) {
+        column = columnInfo.bodyFixedColumns[i];
+        totalFixedColumnsWidth += column.props.width;
+      }
+
+      var scrollableColumnIndex = Math.min(
+        props.scrollToColumn - fixedColumnsCount,
+        columnInfo.bodyScrollableColumns.length - 1,
+      );
+
+      var previousColumnsWidth = 0;
+      for (i = 0; i < scrollableColumnIndex; ++i) {
+        column = columnInfo.bodyScrollableColumns[i];
+        previousColumnsWidth += column.props.width;
+      }
+
+      var availableScrollWidth = props.width - totalFixedColumnsWidth;
+      var selectedColumnWidth = columnInfo.bodyScrollableColumns[
+        scrollableColumnIndex
+      ].props.width;
+      var minAcceptableScrollPosition =
+        previousColumnsWidth + selectedColumnWidth - availableScrollWidth;
+
+      if (scrollX < minAcceptableScrollPosition) {
+        scrollX = minAcceptableScrollPosition;
+      }
+
+      if (scrollX > previousColumnsWidth) {
+        scrollX = previousColumnsWidth;
+      }
+    }
+  }
+
+  var scrollContentWidth =
+    FixedDataTableWidthHelper.getTotalWidth(columns);
+  var horizontalScrollbarVisible = scrollContentWidth > props.width &&
+    props.overflowX !== 'hidden' && props.showScrollbarX !== false;
+  var maxScrollX = Math.max(0, scrollContentWidth - props.width);
+  scrollX = Math.min(scrollX, maxScrollX);
+
+  // isColumnResizing should be overwritten by value from props if avaailble
+  var isColumnResizing = props.isColumnResizing !== undefined ?
+    props.isColumnResizing : oldState && oldState.isColumnResizing;
+
+  return Object.assign({}, oldState, {
```

Comment:
This doesn't deep clone the nested objects which will probably cause issues...
I'll try to cleanup in a later pass